uplink/src/lib.rs

```diff
@@ -450,7 +385,91 @@ impl Uplink {
             })
         });
 
-        Ok(bridge_tx)
+        // Bridge thread to batch data and redicet actions
+        thread::spawn(|| {
+            let rt = tokio::runtime::Builder::new_current_thread()
+                .thread_name("bridge")
+                .enable_time()
+                .build()
+                .unwrap();
+
+            rt.block_on(async move {
+                if let Err(e) = bridge.start().await {
+                    error!("Bridge stopped!! Error = {:?}", e);
+                }
+            })
+        });
+
+        Ok(())
+    }
+
+    pub fn construct_builtins(&mut self, bridge: &mut Bridge) -> Result<BuiltIns, Error> {
+        let bridge_tx = bridge.tx();
+
+        let route = ActionRoute { name: "launch_shell".to_owned(), timeout: 10 };
+        let actions_rx = bridge.register_action_route(route);
```

Comment:
`register_action_route` is sending `event` internally. Is this still required? This was previously required because registration comes from the app.

Reply:
Registration is still happening from outside the bridge, by what other pattern would we symbolize the connection of two ends of the action route?